src/components/ui/coaching-sessions/coaching-notes/extended-link-extension.tsx

```diff
@@ -0,0 +1,92 @@
+import Link from "@tiptap/extension-link";
+
+const LinkWithTitle = Link.extend({
+  addAttributes() {
+    return {
+      ...this.parent?.(),
+      title: {
+        default: null,
+        renderHTML: (attributes) => {
+          if (!attributes.href) {
+            return {};
+          }
+          return {
+            title: `Right-click to open ${attributes.href}`,
+          };
+        },
+      },
+    };
+  },
+});
+
+export const ConfiguredLink = LinkWithTitle.configure({
+  openOnClick: false,
+  autolink: true,
+  defaultProtocol: "https",
+  protocols: ["http", "https"],
+  isAllowedUri: (url, ctx) => {
+    try {
+      // construct URL
+      const parsedUrl = url.includes(":")
+        ? new URL(url)
+        : new URL(`${ctx.defaultProtocol}://${url}`);
+
+      // use default validation
+      if (!ctx.defaultValidate(parsedUrl.href)) {
+        return false;
+      }
+
+      // Placeholder for future disallowed domains if we want to add any
+      // const disallowedProtocols = ["ftp", "file", "mailto"];
+      // const protocol = parsedUrl.protocol.replace(":", "");
+
+      // if (disallowedProtocols.includes(protocol)) {
+      //   return false;
+      // }
+
+      // // only allow protocols specified in ctx.protocols
+      // const allowedProtocols = ctx.protocols.map((p) =>
+      //   typeof p === "string" ? p : p.scheme
+      // );
+
+      // if (!allowedProtocols.includes(protocol)) {
+      //   return false;
+      // }
+
+      // Placeholder for future disallowed domains if we want to add any
+      // const disallowedDomains = [
+      //   "example-phishing.com",
+      //   "malicious-site.net",
+      // ];
+      // const domain = parsedUrl.hostname;
+
+      // if (disallowedDomains.includes(domain)) {
+      //   return false;
+      // }
```

Comment:
Is this still needed? You have implemented some or all of this already from what I can tell.

Reply:
Can definitely remove it. It's not the same as what I've implemented, its more like custom validation that we could absolutely hold off on implementing until its really needed.